lib/ansiblelint/__init__.py

```diff
@@ -101,6 +101,14 @@ def matchtasks(self, file, text):
                            file['path'], self, message))
         return matches
 
+    def _unpack_result(self, play, result):
+        linenumber = play[ansiblelint.utils.LINE_NUMBER_KEY]
```

Comment:
Why do you extract the line number here if you don't need this in one of the branches?

Reply:
I'm trying to lint for loop variables to enforce the loop var policy in zuul-jobs but ended up in a bit of a rabbit hole trying to solve that. matchtask didn't seem to match includes or blocks so I ended up using matchplay instead to solve that, but that makes it so it gives the linenumber of the block containing the task instead of the task the linter fails at.

see:
- https://review.opendev.org/#/c/724910/
- https://review.opendev.org/#/c/724967/

Reply:
Oh, this policy by the way: https://zuul-ci.org/docs/zuul-jobs/policy.html#ansible-loops-in-roles